compiler/noirc_frontend/src/hir/comptime/errors.rs

```diff
@@ -540,6 +546,10 @@ impl<'a> From<&'a InterpreterError> for CustomDiagnostic {
                 let msg = format!("{item} is currently unimplemented");
                 CustomDiagnostic::simple_error(msg, String::new(), location.span)
             }
+            InterpreterError::WillNotImplement { item, location, explanation } => {
+                let msg = format!("{item} is not going to be implemented");
+                CustomDiagnostic::simple_error(msg, explanation.clone(), location.span)
```

Comment:
The `{item}` part of the message is constructed [here](https://github.com/noir-lang/noir/blob/6f47cf14ef4f4d501980ad30f2db711e919ccd26/compiler/noirc_frontend/src/hir/comptime/interpreter/foreign.rs#L78-L92), it can be made separate for the `Unimplemented` and `WillNotImplement` case. Note that there is an `explanation` for the second case which tells that you cannot use X in compile time or we shouldn't be calling Y because it's supposed to be handled differently. 

If you have a better name for the variant then please share. I started with `Unavailable`, then went to `Unsupported`, then realised there are some other `UnsupportedSomethingSomething` already and didn't want to add one that sounds like a more general version of those, so I ended up with the one that conveys exactly what we mean, but I agree the message sounds a bit aggressive.

Reply:
Maybe `InvalidInComptimeContext` for a name?

Reply:
Renamed to `InvalidInComptimeContext` and using different `item`.